packages/sqs-receiver-service/src/receiver.js

```diff
@@ -27,26 +23,20 @@ const receiver = async () => {
   const messages = await readQueue(env.URL, env.VISIBILITY_TIMEOUT_MS, env.WAIT_TIME_MS)
 
   // If we have read any messages then post the body to the subscriber
-  if (messages) {
-    debug('Read %d messages: %O', messages.length, messages)
-
+  if (messages.length) {
     const messageSubscriberResults = await Promise.all(
       messages.map(async m => processMessage(m, env.SUBSCRIBER, env.SUBSCRIBER_TIMEOUT_MS))
     )
-
-    debug({ messageSubscriberResults })
     await deleteMessages(env.URL, messageSubscriberResults)
-
     messageLastReceived = Date.now()
   }
 
   await showQueueStatistics(env.URL)
 
-  // Invoke the poll delay only on a small number of messages processed
-  // if they are coming in thick-and-fast then read again immediately
-  if (!messages) {
+  // Invoke the poll delay only if no messages were received from the queue
+  if (!messages.length) {
```

Comment:
Think this could just be an else on the `if (messages.length) {` above

Reply:
Agreed, change incoming